Add release notes for 16.9.29 (#10910)
chef/chef: RELEASE_NOTES.md

- remove support for RHEL 6 i386 / Ubuntu 16.04
- don't write out node['filesystem2'] data on AIX/Solaris/FreeBSD: https://github.com/chef/ohai/pull/1592

## What's New in 16.9.29

### Chef InSpec 4.25.1

Chef InSpec has been updated from 4.24.8 to 4.25.1:

- OpenSSH Client on Windows can now be tested with the ssh_config and sshd_config resources. Thanks [@rgeissert](https://github.com/rgeissert)!

> - The `--reporter-message-truncation` option now also truncates the `code_desc` field, preventing failures sending large reports to Automate.

Reviewer: ```suggestion
- The `--reporter-message-truncation` option now also truncates the `code_desc` field, preventing failures when sending large reports to Automate.
```

### Bug Fixes

> - Resolved failures running chef-client on some Windows systems.

Reviewer: ```suggestion
- Resolved failures from running `chef-client` on some Windows systems.
```

> - Compliance Phase: Improved detection of the `audit` cookbook being used for compliance reporting.

Reviewer: ```suggestion
- Compliance Phase: Improved detection of the `audit` cookbook when it is used for compliance reporting.
```

- chef-shell: Added support for loading configs in `client.d` directories - Thanks [@jaymzh](https://github.com/jaymzh)!
- Duplicate gems in our packaging have been removed to further shrink the package sizes and improve load time.

## What's New in 16.9.20

- Updated the package resource on FreeBSD to work with recent changes to the pkgng executable. Thanks [@mrtazz](https://github.com/mrtazz/)

> - Added a missing dependency in the chef-zero binary that could cause failures running chef-zero.

Reviewer: ```suggestion
- Added a missing dependency in the chef-zero binary that could cause failures when running chef-zero.
```

> - Resolved failures running the audit cookbook from our yet to be fully released Chef Infra Compliance Phase. As it turns out this dark launch was not as dark as we had hoped.

Reviewer: ```suggestion
- Resolved failures when running the audit cookbook from our yet-to-be-fully-released Chef Infra Compliance Phase. As it turns out, this dark launch was not as dark as we had hoped.
```
